src/rapids_pre_commit_hooks/pyproject_license.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import random
+
+import tomlkit
+import tomlkit.exceptions
+
+from .lint import LintMain
+
+RAPIDS_LICENSE = "Apache 2.0"
+ACCEPTABLE_LICENSES = {
+    "Apache 2.0",
+    "BSD-3-Clause",
+}
+
+
+def find_value_location(document, key):
+    copied_document = copy.deepcopy(document)
+    i = 0
+    while str(placeholder := random.randint(0, 1048576)) in copied_document.as_string():
```

Comment:
I'm super confused about what this is doing. Why are we generating random values to find a string?

Reply:
(I read [this previous thread](https://github.com/rapidsai/pre-commit-hooks/pull/30#discussion_r1586916941) but I still didn't understand the purpose of this.)

Reply:
If we're trying to insert a "garbage" string into the parsed representation in order to find the character index location for that key, we should avoid random integers and instead use a function like [uuid.uuid4()](https://docs.python.org/3/library/uuid.html#uuid.uuid4). We should do this as a one-shot and assume it will work, rather than retrying 10000 times.

Reply:
Yes, that's what we're trying to do. I've replaced the random integer with a UUID as you suggested.